hooks/resolve_publish.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+from sgtk import Hook
+
+
+class PublishResolver(Hook):
+    """
+    Hook used to resolve publish records in Shotgun into
```

Comment:
i think core hooks still make sense! Also note that this can still be overridden in an app hook which is kind of neat - see for example this [loader hook](https://github.com/shotgunsoftware/tk-multi-loader2/blob/master/hooks/tk-maya_actions.py#L144). You can do per-scenario manipulations inside the actual app hooks, but if you wanted to for example implement global logic for `perforce://` one option would be this new core hook.